SpotifySearch/SpotifySearch/AlbumTableViewController.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  AlbumTableViewController.swift
+//  SpotifySearch
+//
+//  Created by Harichandan Singh on 10/28/16.
+//  Copyright © 2016 Harichandan Singh. All rights reserved.
+//
+
+import UIKit
+
+class AlbumTableViewController: UITableViewController {
+    //MARK: - Properties
+    internal var albums: [Album] = []
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let array = Array(APIRequestManager.manager.apiEndpoint.characters)
+        self.title = String(array[36...48])
+        
+        loadAlbums()
+    }
+    
+    func loadAlbums() {
+        APIRequestManager.manager.getData(apiEndpoint: APIRequestManager.manager.apiEndpoint) { (data: Data?) in
+            if data != nil {
+                if let x = Album.createSpotifyObject(from: data!) {
+                    self.albums = x
+                    print("The albums array has been populated!")
+                }
+            }
+            DispatchQueue.main.async {
+                self.tableView.reloadData()
+            }
+        }
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    // MARK: - Table view data source
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return albums.count
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> AlbumTableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "albumCell", for: indexPath) as? AlbumTableViewCell
+        let album = albums[indexPath.row]
+        cell?.titleLabel.text = album.title
+        
+        APIRequestManager.manager.getData(apiEndpoint: album.thumbnailImageString) { (data: Data?) in
+            if let d = data {
+                cell?.thumbnailImage.image = UIImage(data: d)
+            }
+            DispatchQueue.main.async {
```

Comment:
All UI setting must be done on the main thread.

`
DispatchQueue.main.async {
  cell?.thumbnailImage.image = UIImage(data: d)
}
`